src/python/pants/engine/objects.py

```diff
@@ -187,3 +187,10 @@ def of(cls, *element_types):
 
   def __iter__(self):
     return iter(self.dependencies)
+
+  def __bool__(self):
+    return bool(self.dependencies)
+
+  def __nonzero__(self):
+    # For python 2 compatibility.
```

Comment:
Would be good to link to https://github.com/pantsbuild/pants/issues/6071 so that we know to remove this in 1.17.x.

Reply:
Done